nvflare/app_opt/xgboost/histogram_based/executor.py

```diff
@@ -188,6 +239,13 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
             verbose_eval=self.verbose_eval,
         )
 
+        engine = fl_ctx.get_engine()
+        if engine.client.overseer_agent:
+            sp = engine.client.overseer_agent.get_primary_sp()
+            if sp and sp.primary is True:
+                self._server_address = sp.name
+        self.log_info(fl_ctx, f"server address is {self._server_address}")
```

Comment:
since we duplicated this code section, can you wrap this into a method and the re-use it here?

Reply:
the duplicated code also exists in `def handle_event(self, event_type: str, fl_ctx: FLContext):`